### PR TITLE
strtok and value_pairs test to criterion

### PR DESCRIPTION
**Pull request by Kokan**
* Converts the strtok and value_pairs unit test from libtest to criterion.
* Remove not required include from xml parser test.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by lbudai** [comment]
@MrAnno : will you review this PR?

File: lib/value-pairs/tests/test_value_pairs.c [hunk 1]
```diff
@@ -122,85 +127,74 @@ testcase(const gchar *scope, const gchar *exclude, const gchar *expected, GPtrAr
...
   GPtrArray *transformers;
-  GlobalConfig *cfg;
+};
 
-  app_startup();
-  putenv("TZ=MET-1METDST");
-  tzset();
+ParameterizedTestParameters(value_pairs, test)
+{
+  const static struct value_pairs_params params[] =
```
Comment (MrAnno) [inline lib/value-pairs/tests/test_value_pairs.c]: I like the `const` here, I didn't know that `cr_make_param_array` accepts it (Well I checked, it's just a macro that cast the array to a void pointer 😄 ).

Although, you didn't use `const` in `test_strtok_r.c`.

**Pull request merged**